src/components/VirtualizedList/VirtualizedList.tsx

```diff
@@ -1,43 +1,156 @@
-import { useRef, forwardRef, useCallback, useMemo, useEffect, useState } from "react";
-import PropTypes from "prop-types";
+import React, {
+  CSSProperties,
+  ForwardedRef,
+  forwardRef,
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from "react";
 import NOOP from "lodash/noop";
 import cx from "classnames";
-import { VariableSizeList as List } from "react-window";
+import { Layout, ScrollDirection, VariableSizeList as List } from "react-window";
 import AutoSizer from "react-virtualized-auto-sizer";
 import usePrevious from "../../hooks/usePrevious";
 import useThrottledCallback from "../../hooks/useThrottledCallback";
 import useMergeRefs from "../../hooks/useMergeRefs";
 import {
-  getNormalizedItems,
   easeInOutQuint,
   getMaxOffset,
+  getNormalizedItems,
   getOnItemsRenderedData,
   isLayoutDirectionScrollbarVisible
 } from "../../services/virtualized-service";
-import "./VirtualizedList.scss";
 import { ELEMENT_TYPES, getTestId } from "../../utils/test-utils";
+import VibeComponentProps from "src/types/VibeComponentProps";
+import VibeComponent from "../../types/VibeComponent";
+import "./VirtualizedList.scss";
 
-const VirtualizedList = forwardRef(
+type ItemType = {
+  value: string;
```

Comment:
we are sure everything is not optional here?

Reply:
Everything is optional now